app/src/test/java/com/blackboxteam/app/blackbox/UserSessionPersistenceTest.java

```diff
@@ -0,0 +1,54 @@
+package com.blackboxteam.app.blackbox;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+
+import com.blackboxteam.app.blackbox.util.UserSessionPersistence;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+
+/**
+ * Tests for UserSessionPersistenceClass
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class UserSessionPersistenceTest {
+
+    private UserSessionPersistence mMockUserSessionPersistence;
+
+    @Mock
+    Context mMockContext;
```

Comment:
You should initialize this mock before using it. Else, it will throw a null pointer exception.

Reply:
````

@Before
    public void setup(Context context) {
        mMockContext = context;
        mockedSharedPreference = Mockito.mock(SharedPreferences.class);
        mMockUserSessionPersistence = new UserSessionPersistence(mMockContext);
    }
````

`java.lang.Exception: Method setup should have no parameters`

Reply:
!